src/Illuminate/Queue/DatabaseQueue.php

```diff
@@ -184,18 +184,24 @@ protected function buildDatabaseRecord($queue, $payload, $availableAt, $attempts
      *
      * @param  string  $queue
      * @return \Illuminate\Contracts\Queue\Job|null
+     * @throws \Exception|\Throwable
      */
     public function pop($queue = null)
     {
         $queue = $this->getQueue($queue);
 
-        $this->database->beginTransaction();
+        try {
+            $this->database->beginTransaction();
 
-        if ($job = $this->getNextAvailableJob($queue)) {
-            return $this->marshalJob($queue, $job);
-        }
+            if ($job = $this->getNextAvailableJob($queue)) {
+                return $this->marshalJob($queue, $job);
+            }
 
-        $this->database->commit();
+            $this->database->commit();
```

Comment:
I just noticed @sisve comment. If we want to keep the call to `commit` inside `marshalJob`, we need to **not commit** here.

Reply:
marshalJob only executes if we've found a job, we still need to call commit() here to release any locks acquired by getNextAvailableJob

Reply:
Hmm. If `getNextAvailableJob()` returns null the commit in `marshalJob` is not triggered :/

The commit here terminates the transaction correctly if there are no jobs in DB.

Reply:
True that. A little confusing. Would be nice to get **this** merged into 5.5 and another PR that clean this up targeting 5.6 (after this gets merged).

Reply:
@deleugpn thanks that would be great I think.